sdk/python/kfp/components/_component_store.py

```diff
@@ -0,0 +1,94 @@
+__all__ = [
+    'ComponentStore',
+]
+
+from pathlib import Path
+from urllib.parse import urlparse, urljoin
+import warnings
+from . import _components as comp
+
+class ComponentStore:
+    def __init__(self, local_search_paths=None, url_search_prefixes=None):
+        self.local_search_paths = local_search_paths or ['.']
+        self.url_search_prefixes = url_search_prefixes or []
+
+        self._component_file_name = 'component.yaml'
+        self._digests_subpath = 'versions/sha256'
+        self._tags_subpath = 'versions/tags'
+
+
+    def load_component_from_url(self, url):
+        return comp.load_component_from_url(url)
+
+
+    def load_component_from_file(self, path):
+        return comp.load_component_from_file(path)
+
+
+    def load_component(self, name, digest=None, tag=None):
+        '''
+        Loads component local file or URL and creates a task factory function
+                  
+        Search locations:
+        <local-search-path>/<name>/component.yaml
+        <url-search-prefix>/<name>/component.yaml
+
+        If the digest is specified, then the search locations are:
+        <local-search-path>/<name>/versions/sha256/<digest>
+        <url-search-prefix>/<name>/versions/sha256/<digest>
+
+        If the tag is specified, then the search locations are:
+        <local-search-path>/<name>/versions/tags/<digest>
+        <url-search-prefix>/<name>/versions/tags/<digest>
+        
+        Args:
+            name:   Component name used to search and load the component artifact containing the component definition.
+                    Component name usually has the following form: group/subgroup/component
+            digest: Strict component version. SHA256 hash digest of the component artifact file. Can be used to load a specific component version so that the pipeline is reproducible.
+            tag:    Version tag. Can be used to load component version from a specific branch. The version of the component referenced by a tag can change in future.
+
+        Returns:
+            A factory function with a strongly-typed signature.
+            Once called with the required arguments, the factory constructs a pipeline task instance (ContainerOp).
+        '''
+        #This function should be called load_task_factory since it returns a factory function.
+        #The real load_component function should produce an object with component properties (e.g. name, description, inputs/outputs).
+        #TODO: Change this function to return component spec object but it should be callable to construct tasks.
+        if name is None:
```

Comment:
I would do "if not name" so we can also check for empty string.

Reply:
Done.